app/middlewares/auth.js

```diff
@@ -13,24 +13,31 @@ exports.checkCredentials = (req, res, next) => {
   const token = jwt.decode(req.headers.token, config.common.session.secret);
 
   User.findOne({
-    where: token
+    where: {email: token.email}
   }).then(user => {
 
     if(!user){
       return next(errors.invalidCredentialError);
     }
-
+    if(!sessionStillValid(token.creationDate, user.lastInvalidation)){
+      return next(errors.sessionExpired);
+    }
+    
     req.user = user;
     next();
   
   });
 
 };
 
-exports.isAdmin = (req, res, next, user) => {
+exports.isAdmin = (req, res, next) => {
 
   if(!req.user.isAdmin){
     return next(errors.notAnAdmin);
   }
   next();
+};
+
+const sessionStillValid = (creationDate, lastInvalidation) => {
+  return creationDate > lastInvalidation;
```

Comment:
We aware that here you are comparing a timestamp against a Date, for example: `creationDate:  1516386094321 lastInvalidation:  2018-01-19T18:23:12.938Z`, anyways js makes it works without problem.
Also if a user never invalidates his sessions the lastInvalidation will be null, also that kind of works because js takes it as a 0, but it isn't pretty. I think we should at user creation set the lastInvalidation to the current time. What do you think?